### ISSUE TITLE
Spinner - The combination of ariaLive, ariaLabel, and label props necessary for "aria-live" attribute to be present is confusing (or broken?)

### ISSUE THREAD

**Issue by curtisj44** [microsoft/fluentui#14979]
### Environment Information

- **Package version(s)**: 7.137.0 (latest version as of today)
- **Browser and OS versions**: all

### The issue

I want to use the `Spinner` component for status messages with visible text and ARIA attributes necessary for optimal accessibility.

The [`Spinner` documentation](https://developer.microsoft.com/en-us/fluentui/#/controls/web/spinner) indicates that setting `label` and `ariaLive` props is enough to output an `aria-live` politeness level:

![ISpinnerProps interface](https://user-images.githubusercontent.com/810231/92780441-0254b980-f358-11ea-87a3-fc78dc68eaaf.png)
 
### The demo

https://codepen.io/curtisj44/pen/VwaXGEX 📦

### Actual behavior

**Unfortunately, the `ariaLive` prop has no effect on the output when used by itself:**

This JSX:
```jsx
<Spinner
  ariaLive="assertive"
  label="With ariaLive"
/>
```

Results in this HTML:
```html
<div class="ms-Spinner root-43">
  <div class="ms-Spinner-circle ms-Spinner--medium circle-44"></div>
  <div class="ms-Spinner-label label-45">With ariaLive</div>
</div>
```
(Which is identical to the output when no `ariaLive` prop is defined)

**Only when `ariaLabel` is defined as well does `ariaLive` do anything:**

This JSX:
```jsx
<Spinner
  ariaLabel="ARIA label"
  ariaLive="assertive"
  label="With ariaLabel and ariaLive"
/>
```

Results in this HTML:
```html
<div class="ms-Spinner root-43">
  <div class="ms-Spinner-circle ms-Spinner--medium circle-44"></div>
  <div class="ms-Spinner-label label-45">With ariaLabel and ariaLive</div>
  <div role="status" aria-live="assertive">
    <div class="screenReaderText-46">ARIA label</div>
  </div>
</div>
```

### Expected behavior

Based on the [prop descriptions](https://developer.microsoft.com/en-us/fluentui/#/controls/web/spinner#implementation), I was expecting:

1. **The `label` should be announced to screen readers.**

   This description for the `label` props doesn't appear to ever be true:

   > The label to show next to the Spinner. Label updates will be announced to the screen readers. Use ariaLive to control politeness level. 

   (When it is defined, the `ariaLabel` is the text announced to screen readers)

   **Suggestions**

   1. Do the prop description need to be updated? Maybe something like this?

      | Name | Description | 
      | :-- | :-- |
      | ariaLabel | Alternative status label to be announced to the screen readers. Use `ariaLive` to control politeness level. |
      | ariaLive | Politeness setting for label update announcement. Requires a `ariaLabel` to be defined. |
      | label | The label to show next to the Spinner. |

   1. Alternatively, when a `label` prop is defined, should `aria-live` and `role` attributes be automatically output? Is there a valid reason for having a visual label (as defined by `label`) that isn't presented audibly as well? This would avoid the need to define both a `label` and an `ariaLabel` (and create better default accessibility).

1. **The `ariaLive` prop should be usable independently**

    I was expecting this markup:

   ```jsx
   <Spinner
     ariaLive="assertive"
     label="With ariaLive"
   />
   ```

   To output with `aria-live` defined somewhere:

   ```html
   <div class="ms-Spinner root-43">
     <div class="ms-Spinner-circle ms-Spinner--medium circle-44"></div>
     <div class="ms-Spinner-label label-45" aria-live="assertive">With ariaLive</div>
   </div>
   ```

   **Suggestions** 

   If `ariaLive` is intended to only work when `ariaLabel` is present, it would be helpful to:

    1. Indicate this in the prop description
    1. Throw a warning when that expectation is being broken

**Comment by paulgildea** [comment]
@curtisj44 Thanks for the thorough issue - it's really appreciated!

I'm going to mark this a question for now - until we can get a solid answer from the code owner. 

@smhigley FYI because this touches the a11y API

@aneeshack4 / @dzearing - What's the expected behavior of the API? Also is it expected that developers need to fill both the `ariaLabel` and `label` props correct? We don't do anything automatically like propagate the value in `label` prop to the `aria-label` prop correct?

**Comment by dzearing** [comment]
My suggestion here:

When `ariaLive` is provided along with `label` but no `ariaLabel`, we should fall back to use `label` prop value for the live message. If `ariaLabel` is provided, we should use that.

Reasoning would be that we don't really break contracts - if you're providing ariaLabel with ariaLive today, nothing changes. But additionally if your live message is no different from your label, there's no good reason to provide both.

There are certainly cases where an ariaLabel would be provided without a visible label, so I do believe we still need the 2 separate values.

And the comment should be updated accordingly.

@smhigley Sound reasonable? Or do you have any other suggestions?

**Comment by dawolff-ms** [comment]
@dzearing has there been any effort to implement this change?
